fabtests/unit/getinfo_test.c

```diff
@@ -729,6 +729,37 @@ static int test_caps_regression(char *node, char *service, uint64_t flags,
 	return 0;
 }
 
+/* Check if at least one of FI_LOCAL_COMM and FI_REMOTE_COMM is set */
+static int test_comm_caps(char *node, char *service, uint64_t flags,
+		struct fi_info *base_hints, struct fi_info **info)
+{
+	struct fi_info *test_hints = NULL;
+	uint64_t test_caps[2] = { FI_LOCAL_COMM, FI_REMOTE_COMM };
+	int ret, i;
+
+	if (base_hints) {
+		test_hints = fi_dupinfo(base_hints);
+		if (!test_hints)
+			return -FI_ENOMEM;
+	}
+
+	for (i = 0; i < 2; i++) {
+		test_hints->caps &= ~(FI_LOCAL_COMM | FI_REMOTE_COMM);
+		test_hints->domain_attr->caps &= ~(FI_LOCAL_COMM | FI_REMOTE_COMM);
+		test_hints->caps |= test_caps[i];
+		test_hints->domain_attr->caps |= test_caps[i];
+		ret = fi_getinfo(FT_FIVERSION, node, service, flags, test_hints, info);
+		if (ret == 0)
+			goto out;
```

Comment:
We can strengthen this check a little with something like this:
```
((info->caps & test_caps[i]) == test_caps[i]) &&
(((info->caps & test_caps[i]) == ((info->domain_attr->caps & test_caps[i])))
```

Reply:
Done.